### PR TITLE
Adds models for holds and hold_sources

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds models for Holds and Hold_Sources, along with basic tests.

The details of these models are derived from the 1.2 data model that can be found linked to the Jira ticket in this PR.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

YES

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/models/hold.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#  case_number      :string
+#  status           :integer          not null
+#  processing_notes :text
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#
+class Hold < ApplicationRecord
+	belongs_to :thesis
+	belongs_to :hold_source
+
```
Comment (JPrevost) [inline app/models/hold.rb:20]: Adding validations for each required element in the models would be better here and then associated tests to confirm our business rules. The seemingly large amount of tests in this case won't be testing Rails so much as making sure we notice when someone changes the model validations in a way that should break our understanding of the business rules as defined in our tests, we should find out by tests failing.

Right now, I can create a record with no date_end, no date_start, no  date_requested and when I check `.valid?` says true but when I save it it fails with a db constraint error.

```
#<Hold:0x00007fef07641e70
 id: nil,
 thesis_id: 43,
 date_requested: nil,
 date_start: nil,
 date_end: nil,
 hold_source_id: 1,
 case_number: "123456",
 status: "active",
 processing_notes: "Hallo!",
 created_at: Wed, 13 Jan 2021 19:26:44 UTC +00:00,
 updated_at: Wed, 13 Jan 2021 19:26:44 UTC +00:00>
```

☝🏻 that's clearly an invalid Hold by our standards but rails thinks it is valid and will try to save it (and then fail).

File: app/models/hold_source.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# == Schema Information
+#
+# Table name: hold_sources
+#
+#  id         :integer          not null, primary key
+#  source     :text             not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
```
Comment (JPrevost) [inline app/models/hold_source.rb:9]: Similarly to the Holds mode, please add validations to this model to allow Rails to know when an Object is valid and not. You can see why quickly in this model in the `rails console` by:
hs = HoldSource.new
hs.valid?

That will be true. It is not :)

File: test/models/hold_test.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#  updated_at       :datetime         not null
+#
+require 'test_helper'
+
+class HoldTest < ActiveSupport::TestCase
+  test 'valid hold' do
+    hold = holds(:valid)
+    assert(hold.valid?)
+  end
+
```
Comment (JPrevost) [inline test/models/hold_test.rb:24]: See Holds model comment about adding tests for all of the not nullable fields.

File: test/models/hold_source_test.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Table name: hold_sources
+#
+#  id         :integer          not null, primary key
+#  source     :text             not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+require 'test_helper'
+
```
Comment (JPrevost) [inline test/models/hold_source_test.rb:11]: See HoldSource for comment about adding tests for not nullable fields.

**Review by JPrevost — COMMENTED** [review]
See comments about some required changes around model validations and a couple small things that can happen later under the UI ticket or spin into new tickets.

File: app/models/hold_source.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+class HoldSource < ApplicationRecord
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/models/hold_source.rb:10]: Adding the values from the data model to `db/seeds.rb` sounds like a perfect thing for this PR to tackle. I'm still debating whether the admin UI bits should be in this PR or the other UI one...stay tuned there but yes - if it doesn't land here it should be in the other one.

**Comment by matt-bernhardt** [comment]
Okay @JPrevost - I think I've addressed the feedback you provided. Can you take another look now?

**Review by JPrevost — APPROVED** [review]
👍🏻

**Pull request merged**